aten/src/ATen/native/QuantizedLinear.cpp

```diff
@@ -0,0 +1,192 @@
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+#include "ATen/WrapDimUtilsMulti.h"
+
+#ifdef USE_FBGEMM
+#include "fbgemm/Fbgemm.h"
+#include "fbgemm/QuantUtils.h"
+#endif // USE_FBGEMM
+
+#include <array>
+#include <cctype>
+#include <cmath>
+#include <cstddef>
+#include <sstream>
+#include <string>
+#include <vector>
+
+
+#include <chrono>
+namespace at { namespace native {
+
+#ifdef USE_FBGEMM
+
+union PtrPun {
```

Comment:
can we have big red warning of what's going on here and a TODO to type-check it robustly (by enabling custom opaque instead)?

Reply:
This is a WIP diff and shouldn't be reviewed. I don't want to to do it this way in the final PR but I'm using it for testing